### PR TITLE
HPA upgrade

### PR DESCRIPTION
**Pull request by cliveseldon**
Fixes: #4339
Fixes: #3854

This PR investigates upgrading HPA.

Presently we use v2beta1 - [this will no longer be served in 1.25 of k8s](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#horizontalpodautoscaler-v125) This PR initial looks to upgrade to v2 spec which is available from 1.23 of k8s. Note latest [istio1.16 only  supports/tested from 1.22 onwards](https://istio.io/latest/docs/setup/install/) which is another reason to increase min k8s version.

 * Converts current v2beta1 specs to v2 
 * Adds a new CRD field `metricsv2` for users wanting to specify v2 metrics definition
 * Always create v2 HPAs now
 * Updated autoscaling and installation docs (for min versions)

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 7f80224644700b2ef8ad11b4ca8ba2cebc20f1c4 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4172-integration-njf7c) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 7f80224644700b2ef8ad11b4ca8ba2cebc20f1c4 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4172-integration-njf7c) | `/test integration`
notebooks | 7f80224644700b2ef8ad11b4ca8ba2cebc20f1c4 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4172-notebooks-xvp5c) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 7f80224644700b2ef8ad11b4ca8ba2cebc20f1c4 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4172-notebooks-xvp5c) | `/test notebooks`
integration | d4a1b174b84be442fd66c67e1da0bc3d606eba14 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4172-integration-gb88p) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | d4a1b174b84be442fd66c67e1da0bc3d606eba14 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4172-integration-gb88p) | `/test integration`
notebooks | d4a1b174b84be442fd66c67e1da0bc3d606eba14 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4172-notebooks-rpm6w) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by adriangonz** [comment]
Looking at the logs from the integration tests, we may need to update the Kind environment where they are run to use a newer version of K8s?

```
2023-02-03T18:11:57.985330Z	warn	installer	retrieving resources to prune type autoscaling/v2beta1, Kind=HorizontalPodAutoscaler: no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta1" not found
2023-02-03T18:11:57.998849Z	warn	installer	retrieving resources to prune type policy/v1beta1, Kind=PodDisruptionBudget: no matches for kind "PodDisruptionBudget" in version "policy/v1beta1" not found
2023-02-03T18:12:01.266601Z	warn	installer	retrieving resources to prune type networking.istio.io/v1alpha3, Kind=DestinationRule: no matches for kind "DestinationRule" in version "networking.istio.io/v1alpha3" not found
2023-02-03T18:12:01.277901Z	warn	installer	retrieving resources to prune type networking.istio.io/v1alpha3, Kind=EnvoyFilter: no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3" not found
2023-02-03T18:12:01.288075Z	warn	installer	retrieving resources to prune type networking.istio.io/v1alpha3, Kind=Gateway: no matches for kind "Gateway" in version "networking.istio.io/v1alpha3" not found
2023-02-03T18:12:01.818742Z	warn	installer	retrieving resources to prune type networking.istio.io/v1alpha3, Kind=VirtualService: no matches for kind "VirtualService" in version "networking.istio.io/v1alpha3" not found
2023-02-03T18:12:02.460026Z	warn	installer	retrieving resources to prune type security.istio.io/v1beta1, Kind=PeerAuthentication: no matches for kind "PeerAuthentication" in version "security.istio.io/v1beta1" not found
2023-02-03T18:12:03.099244Z	warn	installer	retrieving resources to prune type admissionregistration.k8s.io/v1beta1, Kind=MutatingWebhookConfiguration: no matches for kind "MutatingWebhookConfiguration" in version "admissionregistration.k8s.io/v1beta1" not found
2023-02-03T18:12:03.738783Z	warn	installer	retrieving resources to prune type admissionregistration.k8s.io/v1beta1, Kind=ValidatingWebhookConfiguration: no matches for kind "ValidatingWebhookConfiguration" in version "admissionregistration.k8s.io/v1beta1" not found
```

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | d4a1b174b84be442fd66c67e1da0bc3d606eba14 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4172-notebooks-rpm6w) | `/test notebooks`
integration | 977da33e6adb72506f9c667e28e97ce102234bae | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4172-integration-5xdzk) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 977da33e6adb72506f9c667e28e97ce102234bae | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4172-integration-5xdzk) | `/test integration`
notebooks | bab6380e06d5c710fdf042d44847dbb5b0885347 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4172-notebooks-k8csv) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: operator/apis/machinelearning.seldon.io/v1/conversion.go
Comment (RafalSkolasinski) [inline operator/apis/machinelearning.seldon.io/v1/conversion.go:5]: Does it mean that our logic here will work as long as we rely on version of K8s client that still comes with `k8s.io/api/autoscaling/v2beta1`? And once we upgrade we will have to bring in the relevant `v2beta1` code into the repo/.

**Review by RafalSkolasinski — APPROVED** [review]
LGTM! Few general questions. Suggested changes only applies to docs so we can then merge without re-running integration/notebook tests

**Comment by RafalSkolasinski** [comment]
for record - notebook/integration tests pass 
image

File: doc/source/workflow/install.rst
Comment (cliveseldon) [reply doc/source/workflow/install.rst:7]: No higher ceiling I think as all APIs are now on stable versions

File: operator/apis/machinelearning.seldon.io/v1/conversion.go
Comment (cliveseldon) [reply operator/apis/machinelearning.seldon.io/v1/conversion.go:5]: yes

**Pull request merged**
